packages/origin-247-claim/tests/claim-service.e2e-spec.ts

```diff
@@ -0,0 +1,125 @@
+import { bootstrapTestInstance, userWallet } from './setup-e2e';
+import { INestApplication, Logger } from '@nestjs/common';
+import { MatchResultRepository } from '../src/repositories/MatchResult/MatchResult.repository';
+import { DatabaseService } from '@energyweb/origin-backend-utils';
+import { LeftoverConsumptionRepository } from '../src/repositories/LeftoverConsumption/LeftoverConsumption.repository';
+import { ExcessGenerationRepository } from '../src/repositories/ExcessGeneration/ExcessGeneration.repository';
+import { ClaimService } from '../src/claim.service';
+import { SpreadMatcher } from '../src/algorithms/spreadMatcher';
+import {
+    IConsumption,
+    IGeneration,
+    IMatch,
+    IMatchingInput,
+    IMatchingOutput
+} from '../src/interfaces';
+import { BigNumber } from 'ethers';
+import { CertificateService, IBatchClaimCommand } from '@energyweb/origin-247-certificate';
+
+jest.setTimeout(60 * 1000);
+
+describe('Claiming - e2e', () => {
+    let app: INestApplication;
+    let matchResultRepository: MatchResultRepository;
+    let databaseService: DatabaseService;
+    let leftoverConsumptionRepository: LeftoverConsumptionRepository;
+    let excessGenerationRepository: ExcessGenerationRepository;
+    let claimService: ClaimService;
+    let certificateService: CertificateService;
+
+    beforeAll(async () => {
+        ({
+            app,
+            matchResultRepository,
+            databaseService,
+            leftoverConsumptionRepository,
+            excessGenerationRepository,
+            claimService,
+            certificateService
+        } = await bootstrapTestInstance());
+
+        await app.init();
+    });
+
+    beforeEach(async () => {
+        await databaseService.cleanUp();
+    });
+
+    it('should match and claim', async () => {
```

Comment:
Please add test, that will verify how passing additional metadata alongisde generation and consumption works.

Does it work, is it nice to work with, are there any typing problems?